webtau-core/src/main/java/org/testingisdocumenting/webtau/documentation/DocumentationArtifacts.java

```diff
@@ -23,8 +24,18 @@
 import org.testingisdocumenting.webtau.utils.JsonUtils;
 
 import java.nio.file.Path;
+import java.util.concurrent.ConcurrentHashMap;
 
 public class DocumentationArtifacts {
+    private static final ConcurrentHashMap<String, Boolean> usedArtifactNames = new ConcurrentHashMap<>();
+
+    public static void registerName(String artifactName) {
+        Boolean previous = usedArtifactNames.put(artifactName, true);
```

Comment:
I think a set of names with a contains check before a put would be a little clearer but probably not worth changing now.